scio-core/src/main/scala/com/spotify/scio/io/TextIO.scala

```diff
@@ -56,8 +56,8 @@ final case class TextIO(path: String) extends ScioIO[String] {
   override def tap(params: ReadP): Tap[String] =
     TextTap(ScioUtil.addPartSuffix(path))
 
-  private def textOut(path: String, params: WriteP) =
-    BTextIO
+  private def textOut(path: String, params: WriteP) = {
+    var transform = BTextIO
       .write()
       .to(pathWithShards(path))
```

Comment:
this means that there will be a `part` added to the output path, which means the caller will not be able to fully control the single file name

Reply:
hm not really, `pathWithShards` is actually a misleading name since what it does is to remove a trailing slash. :smile: